The tracks are now sorted and have the "#" attribute

ShadyMedic opened the pull request:
There were some tracks that were already published but were missing in the repository (for example the Discovery Jingle or the series of the Journey Beyond songs); these were downloaded from YouTube (see commit messages for links) and sorted along with others.

Some songs are announced but weren't published yet. To make their future inclusion easier, they are now represented by empty `.mp3` files, their author is set to `Unknown` and their album is set according to [Wynncraft Wiki](https://wynncraft.gamepedia.com/Music). When these are published, they can easily replace the empty files without the need to reindex all the tracks.

About the mechanism of ordering, I've already explained this on Wynncraft forums, so [here is a link](https://forums.wynncraft.com/threads/wynncraft-soundtrack-as-downloadable-playlist-of-mp3s.274877/) with more information. (In case of forum wipe, you can use [this archived copy of the thread](https://archive.vn/3yRYp)).

None of the existing files was renamed, so if their names are used by Wynntils music player, it shouldn't cause any problems. The only thing I can think of is, that the empty files might override game music for those areas. So if it works like this, you might want to rename them to not override the online music. List of empty tracks is mentioned in the commit message of 512dfdcba52d2bd7b80b3d850459f287a72b51db

ShadyMedic committed:
Moved all songs to folders representing 5 levels each

ShadyMedic committed:
Addition of track for discovery jingle (An Interesting Find)

File was downloaded from YouTube https://www.youtube.com/watch?v=P2JpGrvfGMs

ShadyMedic committed:
Addition of track for new song for Pigman's Ravines (Military March)

File was downloaded from YouTube https://www.youtube.com/watch?v=UvpPwjkIjDE

ShadyMedic committed:
Addition of two tracks for the Misadventure on the Sea quest

Two songs:
1. Skippin' Service (downloaded from https://www.youtube.com/watch?v=0Devs3tcsls)
2. Lost at Sea (downloaded from https://www.youtube.com/watch?v=OPmeTb1r7K0)

ShadyMedic committed:
Addition of old track for Desert (Travelling at Dusk)

File was downloaded from YouTube https://www.youtube.com/watch?v=Jnb61NQwWQs

ShadyMedic committed:
Addition of track Nesaak Tundra ultimate discovery (A Family Fractured)

File was downloaded from YouTube https://www.youtube.com/watch?v=tJqV3Z_WXas

ShadyMedic committed:
Addition of three tracks for A Journey Beyond quest

Three songs:
1. The Hidden Horrors (downloaded from https://www.youtube.com/watch?v=AvpAN0U4FdA)
2. A Twisted Fate (downloaded from https://www.youtube.com/watch?v=8hkk3pVclTQ)
3. A Spineless Screech (downloaded from https://www.youtube.com/watch?v=itOmJ4tjjgk)

ShadyMedic committed:
Addition of track Point of No Return (unknown location)

Track downloaded from YouTube: https://www.youtube.com/watch?v=gvGWZw-RcxY

ShadyMedic committed:
Addition of four tracks that haven't been published yet

These tracks are represented by empty (and identical) .mp3 files
When these songs are released, they can easily replace the empty files
The four songs are:
1. Funeral Service ar 10 o'clock (Death's Realm)
2. Prelude to Insanity (Corrupted Dungeons' Entrances)
3. Out Of Time (Corrupted Dungeons 2)
4. A Frightening Find (Silent Expanse Discovery)

ShadyMedic committed:
Textfile containing the list of sorted tracks is added

Maybe it will make displaying the order of the songs easier in GitHub or something...

ShadyMedic committed:
Finally, all tracks are sorted.

All the tracks have number from 1 to 164 in their "#" attribute.
My reasoning for this order can be found in this shared TODO list (as notes)
https://to-do.microsoft.com/sharing?InvitationToken=cI0_-u7TDBk2a3NRmRwX8Qw1BhzaSES9PyCVcRAPb-_wg4vfOtAZ6RPso-VQTQpQg
All in all, the order is based on the likely order an average player hears the songs in the game.

ShadyMedic committed:
Deletion folders created in 43a4a21e527802456e6d0f9727c13383f127914a

Each track now has only one copy with its number in the "music" folder again.

ShadyMedic committed:
Addition of # attribute to the list of metadata

Somebody might want to rewrite that one line.
I just thought I will edit this in case others forget to update it.

ShadyMedic committed:
Renamed the track for Twain Mansion to fix the merge conflict.

ShadyMedic commented:
Not sure why it reports a conflict, I renamed the file to have the same name and did no other changes apart from adding the "#" attribute. 😕

ShadyMedic committed:
Addition of track Funeral Service at 10 O'Clock (Death's Realm)

This track was missing before, the placeholder file was replaced
The track was downloaded from https://youtu.be/WfXgPxEGMro

ShadyMedic commented:
Since this still wasn't merged, I added the track for Death's Realm, which has been published recently [here](https://youtu.be/WfXgPxEGMro). Metadata are in the same format as in other files.

HeyZeer0 commented:
Oh crap! Since you did not ask for a review I was not notified about this PR!
Yeah, this is going to be a bit of a pain to merge because GitHub is not supposed to handle .mp3 files, but I will see what I can do whenever I've free time.

ShadyMedic commented:
Oh, sorry, my bad. I am still not very experienced with GitHub and didn't know that I need to ask for a review.

I know that you'll want ti check everything yourself just to make sure, but here are the exact steps I took when adding songs or setting their # attribute.

- In case the file already existed
  1. Changed nothing but the # attribute
- In case the file didn't exist but was published by XavierEXE on YouTube
  1. Downloaded .mp3 file from youtube (see commit messages for exact url)
  2. Renamed the file to have the same name as the YouTube video
  3. Set the title attribute to the same value as the name of the file
  4. Set the author attribute, according to the person credited in the description of the YouTube video
  5. Set the album attribute to one of the existing values, according to a section the track was assigned into on Wynncraft Wiki
  6. Set the # attribute
- In case the file didn't exist and wasn't published on YouTube
  1. Created an empty `.mp3` file
  2. Set the name of it to the same format as at other files (`NAME (LOCATION)`), according to information from Wynncraft Wiki
  3. Set the title attribute to the same value as the name of the file
  4. Set the author attribute to "Unknown"
  5. Set the album attribute to one of the existing values, according to a section the track was assigned into on Wynncraft Wiki
  6. Appended `[MISSING]` to the end of the name of the file

And for the track for the Twain Mansion, I renamed it to have the same name as on your master branch.

Hope this helps.

4321ba commented:
Hi! I don't know how this is going, but if you are interested in the missing 4 pieces: Corkian Madness (Dwelling walls), Prelude to Insanity, Out of Time (Corrupted dungeons) and A Frightening Find (SE discovery), I exported them from ONBS to mp3, so they are as close as they can be. Also, as a bonus I included the modified version of Energetic encounter played at the Misadventures on the Sea giant rat boss fight.
[wynncraft_missing_pieces.zip](https://github.com/Wynntils/WynncraftOST/files/5798093/wynncraft_missing_pieces.zip)

ShadyMedic commented:
Thanks a lot, I'll definitelly add them. But it'll be very annoying to rearrange the songs order by shifting their numbers by one or two. I'll probably wait with that before the 1.20 music gets released, so I don't have to do it all over again.

By the way, anyone, don't you know any software that would allow me to sort the tracks by dragging them and then export the whole playlist with the # attributes?

ShadyMedic commented:
Also, @4321ba, does the modified version of Energetic encounter really play during the Giant Rat fight? I can't find it on the wiki.
And @HeyZeer0, since that song isn't on wiki, I'm not sure how should I name it and who is the author. Don't you know the official name and author?

ShadyMedic committed:
Addition of track Prelude to Insanity (Corrupted Dungeons' Entrances)

This track was missing before, the placeholder file was replaced
The track was provided by GitHub user 4321ba (https://github.com/4321ba)

ShadyMedic committed:
Addition of track Out Of Time (Corrupted Dungeons 2)

This track was missing before, the placeholder file was replaced
The track was provided by GitHub user 4321ba (https://github.com/4321ba)

ShadyMedic committed:
Addition of track A Frightening Find (Silent Expanse Discovery)

This track was missing before, the placeholder file was replaced
The track was provided by GitHub user 4321ba (https://github.com/4321ba)

ShadyMedic committed:
Textfile with sorted list of tracks updated

Removed the "(missings)" marks from all tracks that were added recently

4321ba commented:
Yeah, that's right, shifting could be really annoying. I have never worked with these tags, I'm using Juk as music player https://apps.kde.org/en/juk and it can edit tags, but I don't think it can do what you need to do.
Yeah, that "remix" does play there but it's not that easy as it seems, there are 4 parts of it, much like how Genesis of the End plays, the 1st part is an intro, which plays only once, then the 2nd part loops while the 1st phase of the boss, the 3rd part while the 2nd phase, and the 4th part is an outro, you can see it here: https://youtu.be/CnZbd5ZgGtg?t=11422 , but I exported it all-in-one, you can't cut this now. Btw, I could provide the parts too @HeyZeer0 if you're ever interested. I'll probably make a Github repo too with NBS files if I'm allowed to do it.
